apps/verification-portal/src/lib/web3Inbox.ts

```diff
@@ -57,20 +60,28 @@ export function initializeInbox() {
   web3AddressStore.subscribe(async () => {
     if (web3InboxClient) {
       const account = getWeb3InboxAccount();
-      console.log(
-        `Inbox address changed to ${account}, reconfiguring inbox client`
-      );
-      const registered = await web3InboxClient.getAccountIsRegistered(account);
-      const subscribed = web3InboxClient.isSubscribedToDapp(account, domain);
-
-      web3InboxRegistered.set(registered);
-      web3InboxSubscribed.set(subscribed);
-
-      console.log(
-        `!!Registered account ${account} ${registered}, subscribed: ${subscribed}`
-      );
-
-      web3InboxClient.setAccount(getWeb3InboxAccount());
+      if (account) {
+        console.log(
+          `Inbox address changed to ${account}, reconfiguring inbox client`
+        );
+        const registered =
+          await web3InboxClient.getAccountIsRegistered(account);
+        const subscribed = web3InboxClient.isSubscribedToDapp(account, domain);
+
+        web3InboxRegistered.set(registered);
+        web3InboxSubscribed.set(subscribed);
+
+        console.log(
+          `!!Registered account ${account} ${registered}, subscribed: ${subscribed}`
+        );
+
+        web3InboxClient.setAccount(account);
+        registerInbox();
+      } else {
+        console.log(
+          'Inbox account changed to no address, clearing inbox client'
+        );
+      }
     } else {
       console.log('Address changed, no inbox client');
```

Comment:
Don't see a clear function called here. Is it correct?

Reply:
Thank you! Yes, Corret it isn't cleared here as I moved it to connection obeserver. Should have renamed it to "should be cleared" to trace if it is happening. Will change for more clear overview

Reply:
Ok, understood, so all good then, thank you.